### PR TITLE
Extended logs for release version

### PR DESCRIPTION
**Pull request by SalavatR**
Hi! Sometimes we need logs in release version

I did it blindly because I can't build the engine

If I understood correctly, it will be enough to declare one of the options in the .appmanifest file
MIN_LEVEL_SEVERITY_DEBUG || MIN_LEVEL_SEVERITY_USER_DEBUG || MIN_LEVEL_SEVERITY_INFO || MIN_LEVEL_SEVERITY_WARNING || MIN_LEVEL_SEVERITY_ERROR || MIN_LEVEL_SEVERITY_FATAL 

like this

    common:
        context:
            defines: ["MIN_LEVEL_SEVERITY_ERROR"]

### CODE REVIEW

File: engine/dlib/src/dlib/log.cpp [hunk 1]
```diff
@@ -72,10 +72,38 @@ static FILE* g_LogFile = 0;
 static dmCustomLogCallback g_CustomLogCallback = 0;
 static void* g_CustomLogCallbackUserData = 0;
 
+static bool LogAvailable()
```
Comment (JCash) [inline engine/dlib/src/dlib/log.cpp:75]: "IsLogAvailable()" is more in line with the current naming.

File: engine/dlib/src/dlib/log.cpp [hunk 2]
```diff
@@ -72,10 +72,38 @@ static FILE* g_LogFile = 0;
...
+{
+    bool allowed = dLib::IsDebugMode();
+#if  MIN_LEVEL_SEVERITY_DEBUG || MIN_LEVEL_SEVERITY_USER_DEBUG || MIN_LEVEL_SEVERITY_INFO || MIN_LEVEL_SEVERITY_WARNING || MIN_LEVEL_SEVERITY_ERROR || MIN_LEVEL_SEVERITY_FATAL
```
Comment (JCash) [inline engine/dlib/src/dlib/log.cpp:78]: Defold defines should be prefixed with "DM_"

Also, I think "MIN_LEVEL" is very non intuitive and not explaining which system they belong to.
Perhaps "DM_LOG_" is a good prefix.

Also, "MIN" implies that there is a "MAX", but there isn't one, so I think we should skip the "MIN" part.

File: engine/dlib/src/dlib/log.cpp [hunk 3]
```diff
@@ -323,6 +351,7 @@ void dmLogInitialize(const dmLogParams* params)
     thread = dmThread::New(dmLogThread, 0x80000, 0, "log");
     g_dmLogServer->m_Thread = thread;
 
+    g_LogLevel = GetInitLogLevel();
```
Comment (JCash) [inline engine/dlib/src/dlib/log.cpp:354]: This is the only real change in this PR, and I don't see a big difference here.

I think you need to explain your use case in further detail.

**Review by JCash — CHANGES_REQUESTED** [review]

**Comment by JCash** [comment]
What is your use case?

**Comment by SalavatR** [comment]
> What is your use case?
> 
> What platform are you targeting?
> E.g. I believe Android and iOS don't want you to spam output in the log files.
> 
> Also, if you want to enable logging in release builds, the proper way would be to use
> 
> ```
> platforms:
>     common:
>         context:
>             excludeLibs: ["engine_release"]
>             libs: ["engine"]
> ```
> 
> However, when I tested it just now, it's not working correectly, since the server adds both "-lengine_release -lengine" and it will use the release version (with logging turned off).
> That to me is the real fix, to make the override system work as intended.

Hi!
I changed some logic and names
Ideas was:
 1) Enable logs in release, because sometimes release and debug versions work differently
 2) Use DM_LOG_SEVERITY_LEVEL_FROM_  to limit spam
 3) Use g_CustomLogCallback and g_CustomLogCallbackUserData to handle logs with any plugin(Sentry, gameanalitycs, etc)
 4) For testing release version use DM_SYSTEM_LOG_ENABLED to view logs in adb, browser console, etc
 
 If i cant use 
 common:
    context:
        defines: ["MIN_LEVEL_SEVERITY_ERROR"]
        
 Which system should i ovverride and where to initialize?
 
 P.S. sorry for my broken english

**Comment by JCash** [comment]
> because sometimes release and debug versions work differently
In which sense do you mean?

Note that your changes aren't really solving your problem, since you've added the defines to the _engine_ code, and those libraries are statically build into libraries. These libraries are not rebuilt when you build using native extensions.

**Comment by SalavatR** [comment]
https://github.com/SalavatR/def_native_logger 
created extention for logs in release)

**Pull request closed without merging**